pkg/apis/build/v1alpha1/build_types.go

```diff
@@ -208,6 +211,50 @@ func (buildSpec *BuildSpec) StrategyName() string {
 	return buildSpec.Strategy.Name
 }
 
+// VulnerabilityIgnoreOptions refers to ignore options for vulnerability scan
+type VulnerabilityIgnoreOptions struct {
+
+	// Issues references the security issues to be ignored in vulnerability scan
+	Issues []string `json:"issues,omitempty"`
+
+	// Severity indicates the severities of security issues to be ignored (comma separated)
+	Severity string `json:"severity,omitempty"`
+
+	// IgnoreUnfixed indicates flag to display only fixed vulnerabilities
+	Unfixed bool `json:"unfixed,omitempty"`
+}
+
+// VulnerabilityScanOptions references the options for vulnerability scanning
+type VulnerabilityScanOptions struct {
+
+	// Enabled indicates whether to run vulnerability scan for image
+	Enabled bool `json:"enabled,omitempty"`
+
+	// FailPush indicates whether to push the image if the vulnerability scan fails
+	FailPush bool `json:"fail,omitempty"`
+
+	// IgnoreOptions refers to ignore options for vulnerability scan
+	IgnoreOptions *VulnerabilityIgnoreOptions `json:"ignore,omitempty"`
+}
+
+var _ pflag.Value = &VulnerabilityScanOptions{}
+
+func (v *VulnerabilityScanOptions) Set(s string) error {
+	return json.Unmarshal([]byte(s), v)
+}
+
+func (v *VulnerabilityScanOptions) String() string {
+	data, err := json.Marshal(*v)
+	if err != nil {
+		panic(err.Error())
+	}
+	return string(data)
+}
+
+func (v *VulnerabilityScanOptions) Type() string {
+	return "vulnerability-scan-settings"
+}
```

Comment:
@SaschaSchwarze0 @qu1queee @adambkaplan Would like to hear your take on this one. The flag value interface is really neat and useful for exactly the use case we have here. The thing I am interested in: I could understand that someone would say this _spoils_ (for lack of a better word) the types definition here. It adds an additional angle to the structures. Is this something that would bug you? I already discussed this with @karanibm6, the alternative here would be to have a wrapper structure in the respective CLI package that would step in and _should_ be able to be used in a very similar way. What are your thoughts?

Reply:
I created a wrapper for `VulnerabilityScanOptions` and removed the flag value interface from the types definitions.